# Basic Data Exploration & Analysis

I will use for analysis next packages:

In [56]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib as mpl
import seaborn as sns

### Let's reading a data

In [24]:
data = pd.read_csv("HR_comma_sep.csv")
data.head()

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.38             0.53               2                   157   
1                0.80             0.86               5                   262   
2                0.11             0.88               7                   272   
3                0.72             0.87               5                   223   
4                0.37             0.52               2                   159   

   time_spend_company  Work_accident  left  promotion_last_5years  sales  \
0                   3              0     1                      0  sales   
1                   6              0     1                      0  sales   
2                   4              0     1                      0  sales   
3                   5              0     1                      0  sales   
4                   3              0     1                      0  sales   

   salary  
0     low  
1  medium  
2  medium  
3     low  
4     low

In [25]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14999 entries, 0 to 14998
Data columns (total 10 columns):
satisfaction_level       14999 non-null float64
last_evaluation          14999 non-null float64
number_project           14999 non-null int64
average_montly_hours     14999 non-null int64
time_spend_company       14999 non-null int64
Work_accident            14999 non-null int64
left                     14999 non-null int64
promotion_last_5years    14999 non-null int64
sales                    14999 non-null object
salary                   14999 non-null object
dtypes: float64(2), int64(6), object(2)
memory usage: 1.1+ MB


Let's check missing data

In [26]:
data.isnull().sum()

satisfaction_level       0
last_evaluation          0
number_project           0
average_montly_hours     0
time_spend_company       0
Work_accident            0
left                     0
promotion_last_5years    0
sales                    0
salary                   0
dtype: int64

We don't have missing data.

Let's check data types.

In [27]:
data.dtypes

satisfaction_level       float64
last_evaluation          float64
number_project             int64
average_montly_hours       int64
time_spend_company         int64
Work_accident              int64
left                       int64
promotion_last_5years      int64
sales                     object
salary                    object
dtype: object

Let's observe what type of departments we have in our data

In [28]:
data['sales'].unique()

array(['sales', 'accounting', 'hr', 'technical', 'support', 'management',
       'IT', 'product_mng', 'marketing', 'RandD'], dtype=object)

We have 10 different departments.

What type of salaries we have in our departments.

In [29]:
data["salary"].unique()

array(['low', 'medium', 'high'], dtype=object)

Other colums contain float and int datatypes. Lets observe summary.

In [30]:
data.describe()

satisfaction_level  last_evaluation  number_project  \
count        14999.000000     14999.000000    14999.000000   
mean             0.612834         0.716102        3.803054   
std              0.248631         0.171169        1.232592   
min              0.090000         0.360000        2.000000   
25%              0.440000         0.560000        3.000000   
50%              0.640000         0.720000        4.000000   
75%              0.820000         0.870000        5.000000   
max              1.000000         1.000000        7.000000   

       average_montly_hours  time_spend_company  Work_accident          left  \
count          14999.000000        14999.000000   14999.000000  14999.000000   
mean             201.050337            3.498233       0.144610      0.238083   
std               49.943099            1.460136       0.351719      0.425924   
min               96.000000            2.000000       0.000000      0.000000   
25%              156.000000            3.000000       0.000000      0.000000   
50%              200.000000            3.000000       0.000000      0.000000   
75%              245.000000            4.000000       0.000000      0.000000   
max              310.000000           10.000000       1.000000      1.000000   

       promotion_last_5years  
count           14999.000000  
mean                0.021268  
std                 0.144281  
min                 0.000000  
25%                 0.000000  
50%                 0.000000  
75%                 0.000000  
max                 1.000000

Here we can observe max spend time for company 10 years and average monthly working hours are 310. One employee can have max 7 projects.

Let's observe how many projects our employee have and compare how many proje

In [31]:
data.groupby(['left',"number_project"])['satisfaction_level'].count()

left  number_project
0     2                  821
      3                 3983
      4                 3956
      5                 2149
      6                  519
1     2                 1567
      3                   72
      4                  409
      5                  612
      6                  655
      7                  256
Name: satisfaction_level, dtype: int64

People who have 2 projects left the company more common than who have 4-7. But who has 3 projects left job rare.

We need to compare satisfaction levels of our employee who left the job and who not. For this, I created two different tables who work in company and not work.

In [46]:
works = data[data["left"]==0]
works.head()

satisfaction_level  last_evaluation  number_project  \
2000                0.58             0.74               4   
2001                0.82             0.67               2   
2002                0.45             0.69               5   
2003                0.78             0.82               5   
2004                0.49             0.60               3   

      average_montly_hours  time_spend_company  Work_accident  left  \
2000                   215                   3              0     0   
2001                   202                   3              0     0   
2002                   193                   3              0     0   
2003                   247                   3              0     0   
2004                   214                   2              0     0   

      promotion_last_5years  sales salary  
2000                      0  sales    low  
2001                      0  sales    low  
2002                      0  sales    low  
2003                      0  sales    low  
2004                      0  sales    low

In [47]:
notWorks = data[data["left"]==1]
notWorks.head()

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.38             0.53               2                   157   
1                0.80             0.86               5                   262   
2                0.11             0.88               7                   272   
3                0.72             0.87               5                   223   
4                0.37             0.52               2                   159   

   time_spend_company  Work_accident  left  promotion_last_5years  sales  \
0                   3              0     1                      0  sales   
1                   6              0     1                      0  sales   
2                   4              0     1                      0  sales   
3                   5              0     1                      0  sales   
4                   3              0     1                      0  sales   

   salary  
0     low  
1  medium  
2  medium  
3     low  
4     low

In [38]:
less5Years = data[data["time_spend_company"]<5]
less5Years.head()

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.38             0.53               2                   157   
2                0.11             0.88               7                   272   
4                0.37             0.52               2                   159   
5                0.41             0.50               2                   153   
6                0.10             0.77               6                   247   

   time_spend_company  Work_accident  left  promotion_last_5years  sales  \
0                   3              0     1                      0  sales   
2                   4              0     1                      0  sales   
4                   3              0     1                      0  sales   
5                   3              0     1                      0  sales   
6                   4              0     1                      0  sales   

   salary  
0     low  
2  medium  
4     low  
5     low  
6     low

In [41]:
more5Years = data[data["time_spend_company"]>=5]
more5Years.head()

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
1                 0.80             0.86               5                   262   
3                 0.72             0.87               5                   223   
7                 0.92             0.85               5                   259   
8                 0.89             1.00               5                   224   
12                0.84             0.92               4                   234   

    time_spend_company  Work_accident  left  promotion_last_5years  sales  \
1                    6              0     1                      0  sales   
3                    5              0     1                      0  sales   
7                    5              0     1                      0  sales   
8                    5              0     1                      0  sales   
12                   5              0     1                      0  sales   

    salary  
1   medium  
3      low  
7      low  
8      low  
12     low